### DISCUSSION TITLE
Pagelayout with Content (Outlet) and context specific sidebar

### DISCUSSION THREAD

**Discussion by KnisterPeter** [remix-run/remix#4374] [Q&A] [answered]
Hi there,

we are thinking about how to structure our layout, routes and data-loaders for an app which has a sidebar which should show context specific information.

This information could be different on all pages, so it's not just a menu with different entries but totally different data.
We still would like to avoid re-rendering the whole layout (header/footer/spaces in between all parts).

What we currently selected so far was a data-fetching per leaf route (which could decide was is the most context specific data to show), and then put the display logic in the `handle` export. That way a component outside of the leaf route (above in the hierarchy) could render that data into the layout while still embedding the `Outlet` next to it in the DOM.

Is this a good approach? Are there other ideas how to handle this case?

Kind regards
Markus

**Comment by kiliman** [comment, marked as answer]
Yes, this is what I would recommend. It's what makes `useMatches` and `handle` so powerful. It gives you control over the entire component tree, and lets you return data where it makes sense.
